frontend/app/actions/TouchpointActionRefiners.scala

```diff
@@ -0,0 +1,118 @@
+package actions
+
+import actions.ActionRefiners.{SubReqWithContributor, SubReqWithSub, SubRequestOrResult, SubRequestWithContributorOrResult, logger}
+import actions.Fallbacks.{memberHome, notYetAMemberOn}
+import com.gu.memsub.{Membership, Subscriber => MemSubscriber}
+import com.gu.memsub.subsv2.SubscriptionPlan
+import com.gu.salesforce.PaidTier
+import play.api.mvc.Results.{InternalServerError, Ok}
+import services.{AuthenticationService, TouchpointBackendProvider}
+import com.gu.memsub.subsv2.reads.ChargeListReads._
+import com.gu.memsub.subsv2.reads.SubPlanReads._
+import com.gu.memsub.subsv2.{Subscription, _}
+import com.typesafe.scalalogging.LazyLogging
+import play.api.mvc.Security.AuthenticatedRequest
+import play.api.mvc._
+import views.support.MembershipCompat._
+
+import scalaz.{-\/, EitherT, OptionT, \/, \/-}
+import scalaz.std.scalaFuture._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scalaz.{-\/, \/, \/-}
+
+class TouchpointActionRefiners(implicit val touchpointBackend: TouchpointBackendProvider, implicit private val ec: ExecutionContext) extends LazyLogging {
+  private def getContributorRequest[A](request: AuthRequest[A]): Future[String \/ Option[SubReqWithContributor[A]]] = {
```

Comment:
I appreciate that this has just been moved from another file, but do we need action refiners related to contributions in this project anymore? I guess we should probably tidy this up in a separate PR though...